WcaOnRails/app/views/persons/_results_by_event.html.erb

```diff
@@ -43,11 +43,26 @@
                   </td>
                   <td class="round"><%= t "rounds.#{result.roundTypeId}.cellName" %></td>
                   <td class="place"><%= result.pos %></td>
-                  <td class="single <%= 'pb' if pb_markers[result.id][:single] %>">
+                  
+                  <% pb_type={} %>
+                  <% if pb_markers[result.id][:single] %>
+                    <% pb_type[:single] = 'pb' %>
+                    <% if !result.regionalSingleRecord.blank? %>
+                      <% pb_type[:single] = pb_type_class_for_result(result.regionalSingleRecord)  # Check which type are the pb (wr,cr,nr) %>
+                    <% end %>
+                  <% end %>
+                  <% if pb_markers[result.id][:average] %>
+                    <% pb_type[:average] = 'pb' %>
+                    <% if !result.regionalAverageRecord.blank? %>
+                      <% pb_type[:average] = pb_type_class_for_result(result.regionalAverageRecord)  # Check which type are the pb (wr,cr,nr) %>
+                    <% end %>
+                  <% end %>
```

Comment:
Since you already have some logic in results_helper and some in here, would it make since to just put all the logic for this within the helper function?

Reply:
Got it! I have change it all to be in the function. Please check it! Thank you!